internal/service/advancedcluster/model_advanced_cluster.go

```diff
@@ -426,6 +426,16 @@ func CheckRegionConfigsPriorityOrderOld(regionConfigs []admin20240530.Replicatio
 	return nil
 }
 
+func flattenPinnedFCV(cluster *admin.ClusterDescription20240805) []map[string]string {
+	if cluster.FeatureCompatibilityVersionExpirationDate == nil { // pinned_fcv is defined in state only if featureCompatibilityVersionExpirationDate is present in cluster response
+		return nil
+	}
+	nestedObj := map[string]string{}
+	nestedObj["version"] = cluster.GetFeatureCompatibilityVersion()
+	nestedObj["expiration_date"] = *conversion.TimePtrToStringPtr(cluster.FeatureCompatibilityVersionExpirationDate)
```

Comment:
```suggestion
	nestedObj := map[string]string{
             "version": cluster.GetFeatureCompatibilityVersion()
             "expiration_date": *conversion.TimePtrToStringPtr(cluster.FeatureCompatibilityVersionExpirationDate)
	}
	
```

Reply:
TimePtrToStringPtr might return nil so * will panic, better to check, or create a TimePtrToString func or similar

Reply:
There is an if just above that ensure FeatureCompatibilityVersionExpirationDate will not be nil. Modified this statement to use TimeToString directly 👍